app/src/main/java/org/fossasia/susi/ai/adapters/viewholders/MapViewHolder.java

```diff
@@ -20,18 +31,72 @@
     @BindView(R.id.map_image)
     public ImageView mapImage;
     @BindView(R.id.location_pointer)
-    public  ImageView pointer;
+    public ImageView pointer;
     @BindView(R.id.background_layout)
     public LinearLayout backgroundLayout;
 
+    public ChatMessage model;
+    public Context currContext;
+    private String TAG = ChatFeedRecyclerAdapter.class.getSimpleName();
+
     /**
      * Instantiates a new Map view holder.
      *
      * @param itemView the item view
      * @param listener the listener
      */
-    public MapViewHolder(View itemView , ClickListener listener) {
-        super(itemView , listener);
+    public MapViewHolder(View itemView, ClickListener listener) {
+        super(itemView, listener);
         ButterKnife.bind(this, itemView);
     }
+
+    public void setView(ChatMessage chatMessage, Context context) {
+        model = chatMessage;
+        currContext = context;
+        if (model != null) {
+            try {
+                final MapHelper mapHelper = new MapHelper(new MapData(model.getLatitude(), model.getLongitude(), model.getZoom()));
+                pointer.setVisibility(View.GONE);
+                Log.v(TAG, mapHelper.getMapURL());
+
+                Picasso.with(currContext).load(mapHelper.getMapURL())
+                        .into(mapImage, new com.squareup.picasso.Callback() {
+                            @Override
+                            public void onSuccess() {
+                                pointer.setVisibility(View.VISIBLE);
+                            }
+
+                            @Override
+                            public void onError() {
+                                Log.d("Error", "map image can't loaded");
+                            }
+                        });
+
+                mapImage.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        /*
+                          Open in Google Maps if installed, otherwise open browser.
+                        */
+
+                        ((MainActivity)currContext).recyclerAdapter.clearSelection();
+                        ((MainActivity)currContext).recyclerAdapter.actionMode.finish();
```

Comment:
This will most likely cause crash. Because if action mode is not initialized. finish will be called on null object giving null pointer exception. Check if action mode is there and finish it then else open google map. 

pseudo code : 
```
on click view {
    if(item == selected) 
        unselect
    else if(item != selected && any other item == selected)
        select this item
    else if (no item selected)
        open google map
}
```
See there was a reason these if else statements were there earlier. It checks if the clicking should select the view or if selected remove it or just open google maps. Don't remove code unnecessarily.
```
- mapViewHolder.mapImage.setOnClickListener(new View.OnClickListener() {
-                    @Override
-                    public void onClick(View v) {
-                        /*
-                          Open in Google Maps if installed, otherwise open browser.
-                        */
-                        if(selectedItems.size() == 0) {
-                            Intent mapIntent;
-                            if (AndroidHelper.isGoogleMapsInstalled(currContext) && mapHelper.isParseSuccessful()) {
-                                Uri gmmIntentUri = Uri.parse(String.format("geo:%s,%s?z=%s", model.getLatitude(), model.getLongitude(), model.getZoom()));
-                                mapIntent = new Intent(Intent.ACTION_VIEW, gmmIntentUri);
-                                mapIntent.setPackage(AndroidHelper.GOOGLE_MAPS_PKG);
-                            } else {
-                                mapIntent = new Intent(Intent.ACTION_VIEW);
-                                mapIntent.setData(Uri.parse(mapHelper.getWebLink()));
-                            }
-                            currContext.startActivity(mapIntent);
-                        } else {
-                            toggleSelectedItem(position);
-                        }
-                    }
-                });
```